src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.cpp

```diff
@@ -7,9 +7,15 @@ void robotsInFriendlyHalf(std::shared_ptr<World> world_ptr,
 {
     for (auto robot : world_ptr->friendlyTeam().getAllRobots())
     {
-        if (!world_ptr->field().pointInFriendlyHalf(robot.position()))
-        {
-            FAIL() << "Robot " + std::to_string(robot.id()) + " entered enemy half";
-        }
+        robotInFriendlyHalf(robot, world_ptr, yield);
+    }
+}
+
+void robotInFriendlyHalf(Robot robot, std::shared_ptr<World> world_ptr,
+                         ValidationCoroutine::push_type& yield)
+{
+    if (!world_ptr->field().pointInFriendlyHalf(robot.position()))
+    {
+        FAIL() << "Robot " + std::to_string(robot.id()) + " entered enemy half";
```

Comment:
The robot here is a copy that won't get updated, so this validation function doesn't work. We need to pass in a robot_id and look it up every time

Reply:
Fixed, used robot_id instead